app/server/utils/user.py

```diff
@@ -590,3 +590,36 @@ def send_phone_verification_message(to_phone, one_time_code):
         reciever_message = 'Your Sempo verification code is: {}'.format(one_time_code)
 
         send_message(to_phone, reciever_message)
+
+
+def change_pin(user: User, new_pin):
+    user.pin = new_pin
+    db.session.commit()
+    send_pin_change_messages(user, new_pin)
+
+
+def change_initial_pin(user: User, new_pin):
+    user.pin = new_pin
+    user.is_activated = True
+    # TODO [Philip]: Find out whether reset token should be deleted.
+    db.session.commit()
+    send_pin_change_messages(user, new_pin)
+
+
+def send_pin_change_messages(user: User, new_pin):
+    # TODO [Philip]: Find out if there is a better way to check whether commit succeeded
+    if user.pin == new_pin:
```

Comment:
hmm i don't think if commit failed they wouldn't match. i think it'd probably error out, so we'd want to rescue? @enjeyw probably knows better
also if it's salted, string compare like this wouldn't work

Reply:
u should just be able to use the `verify_password` method here.